pkg/jsonschema/generate_config_docs.go

```diff
@@ -0,0 +1,270 @@
+package jsonschema
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/iancoleman/orderedmap"
+	"github.com/jesseduffield/lazycore/pkg/utils"
+	"github.com/samber/lo"
+
+	"gopkg.in/yaml.v3"
+)
+
+type Node struct {
+	Name        string
+	Description string
+	Default     any
+	Children    []*Node
+}
+
+const (
+	IndentLevel                  = 2
+	DocumentationCommentStart    = "<!-- START CONFIG YAML: AUTOMATICALLY GENERATED with `go generate ./..., DO NOT UPDATE MANUALLY -->\n"
+	DocumentationCommentEnd      = "<!-- END CONFIG YAML -->"
+	DocumentationCommentStartLen = len(DocumentationCommentStart)
+)
+
+func insertBlankLines(buffer bytes.Buffer) bytes.Buffer {
+	lines := strings.Split(strings.TrimRight(buffer.String(), "\n"), "\n")
+
+	var newBuffer bytes.Buffer
+
+	previousIndent := -1
+	wasComment := false
+
+	for _, line := range lines {
+		trimmedLine := strings.TrimLeft(line, " ")
+		indent := len(line) - len(trimmedLine)
+		isComment := strings.HasPrefix(trimmedLine, "#")
+		if isComment && !wasComment && indent <= previousIndent {
+			newBuffer.WriteString("\n")
+		}
+		newBuffer.WriteString(line)
+		newBuffer.WriteString("\n")
+		previousIndent = indent
+		wasComment = isComment
+	}
+
+	return newBuffer
+}
+
+func prepareMarshalledConfig(buffer bytes.Buffer) []byte {
+	buffer = insertBlankLines(buffer)
+
+	// Remove all `---` lines
+	strData := buffer.String()
+	lines := strings.Split(strData, "\n")
+
+	var newBuffer bytes.Buffer
+
+	for _, line := range lines {
+		if strings.TrimSpace(line) != "---" {
+			newBuffer.WriteString(line + "\n")
```

Comment:
This is a potential memory allocation, and since we are using a `bytes.Buffer` which is supposed to have smart buffering, it's more efficient to write the two things separately. Changed in fa160096613.

Reply:
Good catch!